### PR TITLE
feat: adds option for ordered list native decimal style

### LINKED ISSUE #7027: [OrderedList] - custom Carbon counter prevents native ol styles from starting in order, Firefox only
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

We are running into an issue in the Gatsby Theme (https://github.com/carbon-design-system/gatsby-theme-carbon/issues/949) where the ordered lists are starting at a random number in Firefox. 

A while back we had some issues with ol and realized that Carbon styles don't use the native `decimal` styles for ordered list, but rather have a `::before` element for the numbers/letters. 

Screen Shot 2020-10-12 at 3 16 04 PM
Screen Shot 2020-10-12 at 3 16 13 PM

We needed to use the native styles for a specific feature (https://github.com/carbon-design-system/gatsby-theme-carbon/pull/893), and overrode the Carbon styles. Later, we realized there's still a custom style that is breaking lists on FF: `counter-reset: item`. We tried overriding the style (see screenshots below), but overriding doesn't work, rather removing the style completely does. 

We need an option to use native ordered list decimals (without the custom counter, and `::before` element). Currently, there is no workaround for this bc overriding the styles doesn't work.

> Is this issue related to a specific component?

Ordered list

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Be able to use native ordered list without pseudo elements and custom counters

> What browser are you working in?

Firefox

> What version of the Carbon Design System are you using?

latest

## Steps to reproduce the issue

1. Notice the ordered list starting at 7, this only happens on Firefox. It's coming from the style `counter-reset: item`
Screen Shot 2020-10-12 at 3 00 37 PM

2. Trying to override the style doesn't do anything
Screen Shot 2020-10-12 at 3 01 17 PM
Screen Shot 2020-10-12 at 3 01 39 PM

3. The only way for it to work correctly is if the style is removed completely from the carbon stylesheet
Screen Shot 2020-10-12 at 3 00 54 PM

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #7027 

This new style is meant to be optional for users that want native decimal styles and not a custom counter for ordered lists. I realize, this means the style won't be left-aligned like the custom counter, hence it being optional. The reason this is important for the Gatsby Theme is because we use the `start` attribute for ordered lists to be able to temporarily end and continue lists as needed. However, this attribute doesn't work with the `counter()` because it needs native decimal styles. Right now we are overriding the list styles, but still have some existing issues with it due to the custom counter (see issue).

#### Changelog

- adds ` .#{$prefix}--list--ordered_native` className for decimal styles which won't use any of the pseudo element styles
- adds example in vanilla dev environment and react storybook under "native styles", with additional list items so reviews can see how the alignment changes in double digit numbers

Wasn't sure if the className or prop name would be intuitive enough, but I'm open for feedback! 🙂 🙂

### CODE REVIEW

**Review by dakahn — APPROVED** [review]

**Comment by dakahn** [comment]
ah missed the busted test and went ahead an approved! @jnm2377

**Comment by aagonzales** [comment]
It looks fine in storybook but is there a way for me to see that its working correctly in the gatsby theme?

**Comment by jnm2377** [comment]
@aagonzales the only real way to test it out on the gatsby theme is by manipulating styles in the dom rn, otherwise you'll only see it when we update the carbon dependencies.

**Review by aagonzales — APPROVED** [review]
Ok blindly approving then!

File: packages/components/tests/styles-test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const files = glob.sync('**/*.scss', {
 const render = promisify(sass.render);
 
 describe('styles', () => {
-  jest.setTimeout(20000);
+  jest.setTimeout(40000);
```
Comment (jnm2377) [inline packages/components/tests/styles-test.js:25]: for reviewers: tests were failing bc it said it was timing out, so I updated it to have more time

File: packages/components/src/components/list/_list.scss [hunk 2]
```diff
@@ -20,13 +20,18 @@
 @mixin lists {
   .#{$prefix}--list--nested,
   .#{$prefix}--list--unordered,
-  .#{$prefix}--list--ordered {
+  .#{$prefix}--list--ordered,
+  .#{$prefix}--list--ordered_native {
```
Comment (joshblack) [inline packages/components/src/components/list/_list.scss]:
```suggestion
  .#{$prefix}--list--ordered--native {
```

To follow our BEM convention, would it make sense to do `list--ordered--native`?

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by joshblack — APPROVED** [review]

**Pull request merged**